analyses/copy_number_consensus_call/src/scripts/compare_variant_calling_updated.py

```diff
@@ -145,115 +108,183 @@
+                    chrom_end = min(fin_end_list2,end_list1)
 
-                ## Format the output consensus CNV
-                ## Column 4 is manta's raw coordinates
-                ## Column 5 is cnvkit's raw coordinates
-                ## Column 6 is freec's raw coordinates
+                    ## Format the output consensus CNV
+                    ## Column 4 is manta's raw coordinates
+                    ## Column 5 is cnvkit's raw coordinates
+                    ## Column 6 is freec's raw coordinates
 
-                ## if jval == 0 (manta) and kval == 1 (cnvkit), put info in column 4 and 5, 6th column is null
-                if jval == 0 and kval == 1:
-                    overlap_chrom = [m[0],str(chrom_start),str(chrom_end),str(cnv_list1).strip(','),list2_chr_str_end.strip(','),'NULL',m[-1]]
+                    ## if the files input are MANTA and CNVKIT, put info in column 4 and 5, 6th column is null
+                    if (list1_name == 'manta' and list2_name == 'cnvkit') or (list1_name == 'cnvkit' and list2_name == 'manta'):
+                        overlap_chrom = [chr_list1,str(chrom_start),str(chrom_end),str(cnv_list1).strip(','),list2_chr_str_end.strip(','),'NULL',m[-1]]
 
-                ## if jval == 0 (manta) and kval == 2 (freec), put info in column 4 and 6, 5th column is null
-                elif jval == 0 and kval == 2:
-                    overlap_chrom = [m[0],str(chrom_start),str(chrom_end),str(cnv_list1).strip(','),'NULL',list2_chr_str_end.strip(','),m[-1]]
+                    ## if the files input are MANTA and FREEC, put info in column 4 and 6, 5th column is null
+                    elif (list1_name == 'manta' and list2_name == 'freec') or (list1_name == 'freec' and list2_name == 'manta'):
+                        overlap_chrom = [chr_list1,str(chrom_start),str(chrom_end),str(cnv_list1).strip(','),'NULL',list2_chr_str_end.strip(','),m[-1]]
+
+                    ## if the files input are CNVKIT and FREEC, put info in column 5 and 6, 4th column is null
+                    elif (list1_name == 'cnvkit' and list2_name == 'freec') or (list1_name == 'freec' and list2_name == 'cnvkit'):
+                        overlap_chrom = [chr_list1,str(chrom_start),str(chrom_end),'NULL',str(cnv_list1).strip(','),list2_chr_str_end.strip(','),m[-1]]
+
+                    ## Add the formatted consensus CNV into a growing list of consensus CNVs
+                    consensus_list = consensus_list + [overlap_chrom]
+
+    return consensus_list
+
+
+## Function to save output files
+def save_to_file(output_file_content, output_path, sample_name):
+    ## Open up the file to write in it, the 'w' option overwrites the file if the file exists.
+    with open( output_path, 'w') as file:
+        ## Loop through the output file and print line by line
+        for k in output_file_content:
+
+            ## Add the sample name and file name to each line
+            single_name = output_path.split('/')[-1]
```

Comment:
Move this line outside the loop, as the path never changes.
you can also replace it with the safer `os.path.basename(output_path)` to avoid separator variation.